src/Common/src/System/Net/HttpDateParser.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Globalization;
+
+namespace System.Net
+{
+    internal static class HttpDateParser
+    {
+        private static readonly string[] s_dateFormats = new string[] {
+            "r", // RFC 1123, required output format but too strict for input
+            "ddd, d MMM yyyy H:m:s 'GMT'", // RFC 1123 (r, except it allows both 1 and 01 for date and time)
+            "ddd, d MMM yyyy H:m:s 'UTC'", // RFC 1123, UTC
+            "ddd, d MMM yyyy H:m:s", // RFC 1123, no zone - assume GMT
+            "d MMM yyyy H:m:s 'GMT'", // RFC 1123, no day-of-week
+            "d MMM yyyy H:m:s 'UTC'", // RFC 1123, UTC, no day-of-week
+            "d MMM yyyy H:m:s", // RFC 1123, no day-of-week, no zone
+            "ddd, d MMM yy H:m:s 'GMT'", // RFC 1123, short year
+            "ddd, d MMM yy H:m:s 'UTC'", // RFC 1123, UTC, short year
+            "ddd, d MMM yy H:m:s", // RFC 1123, short year, no zone
+            "d MMM yy H:m:s 'GMT'", // RFC 1123, no day-of-week, short year
+            "d MMM yy H:m:s 'UTC'", // RFC 1123, UTC, no day-of-week, short year
+            "d MMM yy H:m:s", // RFC 1123, no day-of-week, short year, no zone
+
+            "dddd, d'-'MMM'-'yy H:m:s 'GMT'", // RFC 850
+            "dddd, d'-'MMM'-'yy H:m:s 'UTC'", // RFC 850, UTC
+            "dddd, d'-'MMM'-'yy H:m:s zzz", // RFC 850, offset
+            "dddd, d'-'MMM'-'yy H:m:s", // RFC 850 no zone
+            "ddd MMM d H:m:s yyyy", // ANSI C's asctime() format
+
+            "ddd, d MMM yyyy H:m:s zzz", // RFC 5322
+            "ddd, d MMM yyyy H:m:s", // RFC 5322 no zone
+            "d MMM yyyy H:m:s zzz", // RFC 5322 no day-of-week
+            "d MMM yyyy H:m:s", // RFC 5322 no day-of-week, no zone
+        };
+
+        private static readonly IFormatProvider s_formatProvider = DateTimeFormatInfo.InvariantInfo;
+        private static readonly DateTimeStyles s_style = DateTimeStyles.AllowWhiteSpaces | DateTimeStyles.AssumeUniversal;
```

Comment:
This doesn't need to be in a static field; doing so makes it worse.  It can just be passed at the call site.

Reply:
ok